src/Init/Data/Nat/Bitwise/Lemmas.lean

```diff
@@ -60,6 +60,13 @@ noncomputable def div2Induction {motive : Nat → Sort u}
   unfold bitwise
   simp
 
+@[simp] theorem one_and_eq_mod_two (n : Nat) : 1 &&& n = n % 2 := by
+  match Nat.decEq n 0 with
+  | isTrue n0 => subst n0; decide
+  | isFalse n0 =>
+    simp only [HAnd.hAnd, AndOp.and, land]
+    cases mod_two_eq_zero_or_one n with | _ h => simp [bitwise, n0, h]
```

Comment:
Would this proof work with `if`, as in:
```suggestion
  if n0 : n = 0 then
    subst n0; decide
  else
    simp only [HAnd.hAnd, AndOp.and, land]
    cases mod_two_eq_zero_or_one n with | _ h => simp [bitwise, n0, h]
```
This matches the proof strategy of `and_one_is_mod` and is IMO much more readable